example/plugin/main.go

```diff
@@ -0,0 +1,120 @@
+package main
+
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Author: Michael Adler <michael.adler@siemens.com>
+ */
+
+import (
+	"log"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/siemens/wfx/generated/plugin"
+	"github.com/siemens/wfx/generated/plugin/client"
+	"github.com/siemens/wfx/middleware/plugin/ioutil"
+)
+
+const queueSize = 64
+
+type QueueEntry struct {
+	start   time.Time
+	request *plugin.PluginRequestT
+}
+
+func main() {
+	// NOTE: The working directory for this process is inherited from the wfx parent process
+	file, err := os.OpenFile("plugin.log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0o644)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+	log.SetOutput(file)
+	log.Println("[INFO] Plugin starting...")
+
+	queue := make(chan QueueEntry, queueSize)
+
+	// reader goroutine: it's only purpose is to read from stdin as soon as
+	// something comes in, to prevent the other side from blocking
+	go func() {
+		for {
+			req, err := ioutil.ReadRequest(os.Stdin)
+			if err != nil {
+				log.Println("[ERROR] Failed to receive message", err)
+				continue
+			}
+
+			entry := newQueueEntry(req)
+			// this is important: we rather discard the message instead of
+			// blocking, since we want to read the next message from stdin
+			select {
+			case queue <- entry:
+				log.Println("[DEBUG] Message enqueued")
+				// success
+			default:
+				log.Println("[ERROR] Queue full. Message discarded.")
+				// be nice and inform wfx about it
+				resp := plugin.PluginResponseT{
+					Cookie: req.Cookie,
+					Payload: &plugin.PayloadT{
+						Type: plugin.Payloadgenerated_plugin_client_Response,
+						Value: &client.ResponseT{
+							Status:  client.ResponseStatusDeny,
```

Comment:
In this case, it's denied which is consistent with the design goals. Is/can the wfx-client be informed that this is (hopefully/probably) transient? The body tells so but that needs to be parsed by the client....

Reply:
Well I could introduce a custom status for transient errors, however we would have to map this to an existing HTTP status code. We could use internal server error (500) instead of permission denied (403). What do you think?

Reply:
Hm, 500 Internal Server Error is too harsh, that doesn't imply the transient nature. What about 408 Request Timeout, 503 Service Unavailable, or 504 Gateway Timeout? I think a 5xx code would suit better since it's a potentially transient (internal) wfx server "error". What do you think?

Reply:
Yes sounds good, I have pushed an update.